cost-analyzer/templates/cost-analyzer-deployment-template.yaml

```diff
@@ -412,8 +412,10 @@ spec:
                   key: prometheus-server-endpoint
             - name: CLOUD_PROVIDER_API_KEY
               value: "AIzaSyDXQPG_MHUEy9neR7stolq6l0ujXmjJlvk" # The GCP Pricing API requires a key.
+            {{- if .Values.kubecostProductConfigs.gcpSecretName }}
             - name: GOOGLE_APPLICATION_CREDENTIALS
               value: /var/configs/key.json
+            {{- end }}
```

Comment:
For Kubecost folks (@nealormsbee and @mbolt35): Does the GCP credential flow in the frontend just update a mounted secret at `/var/configs/key.json`? If so, I think this Helm chart change will probably break the frontend-only GCP integration flow because the `GOOGLE_APPLICATION_CREDENTAILS` env var won't be set because `kubecostProductConfigs.gcpSecretName` will be `null` (the default).

I know the configuration matrix here is always pretty tricky. What do you think about a new config var `kubecostProductConfigs.gcpUseWorkloadIdentity` which would disable the `GOOGLE_APPLICATION_CREDENTIALS` env var. This preserves our default behavior while still enabling Workload Identity users.

Reply:
Also @Sean-Holcomb, actually! I just realized you were active on the other PR -- if you have a chance, I'd love your input here.

Reply:
This change is fine in general, and wont break current integration methods, however this alone is insufficient to allow bigquery integration via WI, I have an enhancement request open for this in our closed source repository.

Reply:
Hey, 

This would be amazing to have, even to be able to point KC at my GCP account without leaking a json file. 

Is there an argument for going with something that works mostly now?